airbyte-cdk/bulk/toolkits/load-avro/src/testFixtures/kotlin/io/airbyte/cdk/load/data/avro/AvroRecordToAirbyteValue.kt

```diff
@@ -33,19 +34,25 @@ import org.apache.avro.generic.GenericRecord
 import org.apache.avro.util.Utf8
 
 class AvroRecordToAirbyteValue {
-    fun convert(avroValue: Any?, schema: AirbyteType): AirbyteValue {
+    fun convert(avroValue: Any?, schema: AirbyteType, depth: Int): AirbyteValue {
+        if (avroValue == null) {
+            return NullValue
+        }
         when (schema) {
             is ObjectType -> {
                 val properties = LinkedHashMap<String, AirbyteValue>()
                 schema.properties.forEach { (name, field) ->
                     val value = (avroValue as GenericRecord).get(name)
-                    properties[name] = convert(value, field.type)
+                    if ((value != null) || depth < 2) {
```

Comment:
I don't think I understand - if the value is nonull, or the value is null but we're at depth 0 or 1, we set the value? what happens if we find a null at deeper depth?

Reply:
The point is to make there not be null values at the top level so the tests work

If I change them to all null or all unset then other tests break (I think the json tests?)